packages/@n8n/ai-workflow-builder.ee/evaluations/utils/score.ts

```diff
@@ -3,20 +3,44 @@ import type { ProgrammaticEvaluationResult, SingleEvaluatorResult } from '@/vali
 export function calculateOverallScore(
 	evaluatorResults: Omit<ProgrammaticEvaluationResult, 'overallScore'>,
 ): number {
-	const categories = Object.keys(evaluatorResults) as Array<keyof typeof evaluatorResults>;
-
-	const weights: Record<keyof typeof evaluatorResults, number> = {
+	// Base weights for when similarity is not available
+	const baseWeights = {
 		connections: 0.25,
 		trigger: 0.25,
 		agentPrompt: 0.2,
 		tools: 0.2,
 		fromAi: 0.1,
 	};
 
-	const total = categories.reduce(
-		(acc, category) => acc + evaluatorResults[category].score * weights[category],
-		0,
-	);
+	// If similarity is available, adjust weights to include it
+	let weights: Record<string, number>;
+	let applicableCategories: string[];
+
+	if (evaluatorResults.similarity) {
+		// Rebalance weights to include similarity (20% weight)
+		weights = {
+			connections: 0.2,
+			trigger: 0.2,
+			agentPrompt: 0.15,
+			tools: 0.15,
+			fromAi: 0.1,
```

Comment:
Nit: The scoring change here means that `fromAi` is now a bit more important than it used to be, would it make sense to adjust the weights as such:
```
connections: 0.2,
trigger: 0.2,
agentPrompt: 0.16,
tools: 0.16,
fromAi: 0.08,
similarity: 0.2,
```